File: 2021/10/02/RotateString.java
/**
 * https://leetcode.com/problems/rotate-string/
 * @author Andrew Jarombek
 * @since 10/2/2021
 */

public class RotateString {
    public boolean rotateString(String s, String goal) {
        String str = s + s;
        return s.length() == goal.length() && str.contains(goal);
    }
}

File: 2023/01/02/rotateString.js
/**
 * https://leetcode.com/problems/rotate-string/
 * @author Andrew Jarombek
 * @since 1/2/2023
 */

const rotateString = function(s, goal) {
    if (s.length !== goal.length) {
        return false;
    }

    return (s + s).includes(goal);
}

console.assert(rotateString("abcde", "cdeab"));
console.assert(!rotateString("abcde", "abced"));
